### PR TITLE
[WIP] Add reverse_sequence op

### PR DESCRIPTION
**Pull request by farizrahman4u**
Trying to figure out adding new ops to nd4j.

### CODE REVIEW

**Comment by farizrahman4u** [comment]
@eraly SameDiff test passing, please check TF mapping.

File: nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/transforms/ReverseSequence.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    @Override
+    public String onnxName() {
+        throw new NoOpNameFoundException("No onnx op opName found for " + opName());
+    }
+
+    @Override
+    public String tensorflowName() {
+        throw new NoOpNameFoundException("ReverseSequence");
```
Comment (eraly) [inline nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/transforms/ReverseSequence.java]: return "ReverseSequence";

**Comment by eraly** [comment]
@agibsonccc Needs to map int byte buffers to floating point
https://github.com/deeplearning4j/nd4j/blob/6a7efaa05f1b3c5d2f8177cf34be3da460e48acd/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/imports/graphmapper/tf/TFGraphMapper.java#L960

To test pull reverse_sequence branch in nd4j and run TFGraphTestSubDir.java
Don't forget to pull dl4j-test-resources and build again

**Comment by eraly** [comment]
Import test now passes with
https://github.com/deeplearning4j/nd4j/pull/2766/commits/26eecb89d20c3948c1ff822c8c132a10975f0125

File: nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/transforms/ReverseSequence.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    public Map<String, Map<String, PropertyMapping>> mappingsForFunction() {
+        Map<String, Map<String, PropertyMapping>> ret = new HashMap<>();
+        Map<String, PropertyMapping> attrs = new LinkedHashMap<>();
+        val seqDim = PropertyMapping.builder()
+                .propertyNames(new String[]{"seqDim"})
+                .tfInputPosition(2)
+                .build();
+        val batchDim = PropertyMapping.builder()
+                .propertyNames(new String[]{"batchDim"})
+                .tfInputPosition(3)
```
Comment (eraly) [inline nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/transforms/ReverseSequence.java]: This should map to attrs.  .tfAttrName("seq_dim") and .tfAttrName("batch_dim"). The ops.proto file will list what is an attr and what is an input.

**Comment by eraly** [comment]
This + the changes in the TFGraphMapper is now open as https://github.com/deeplearning4j/nd4j/pull/2766. Closing this.

**Pull request closed without merging**